### PR TITLE
SPM-33 Retrieve Enrolled Courses

### PR DESCRIPTION
**Pull request by brandontjd**
# Summary

Created Official Enrollment files. Main files to check for PR:

1. `test_official_enroll.py`
2. `models/official_enroll.py`
3. `api/schemas/official_enroll.py`
4. `api/resources/official_enroll.py`

# Notes

As per the ticket in SPM-33, the main focus is only on returning all official enrolment based on engineer id, inside of `api/resources/official_enroll.py` there is an extra Resource, but ignore that for now.

### CODE REVIEW

File: restful_api/myapi/manage.py [hunk 1]
```diff
@@ -14,11 +14,11 @@ def init():
...
-    click.echo("create user")
-    user = User(username="testuser", email="testuser@mail.com", password="testpassword", active=True)
-    db.session.add(user)
-    db.session.commit()
-    click.echo("created user admin")
+    # click.echo("create user")
+    # user = User(username="testuser", email="testuser@mail.com", password="testpassword", active=True)
+    # db.session.add(user)
+    # db.session.commit()
+    # click.echo("created user admin")
```
Comment (brandontjd) [inline restful_api/myapi/manage.py:21]: I forgot to uncomment this. I originally commented this out as `manage.py` isn't v useful in development phase due to isolation from pytest.

File: restful_api/tests/factories.py [hunk 2]
```diff
@@ -25,11 +25,22 @@ class EmployeeFactory(factory.Factory):
...
     user_type = factory.Sequence(lambda n: "user type %d" % n)
-    
     class Meta:
         model = Employee
 
+class OfficialEnrollFactory(factory.Factory):
+    eng_id = factory.Sequence(lambda n: 1)
+    course_id = factory.Sequence(lambda n: "%d" % n)
+    # start_date = factory.Sequence(lambda n: "%d" % n)
+    # end_date = factory.Sequence(lambda n: "%d" % n)
```
Comment (brandontjd) [inline restful_api/tests/factories.py:35]: Left out testing of start and end date, not important for now.

File: restful_api/myapi/api/schemas/official_enroll.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from myapi.extensions import ma, db
+
+
+class OfficialEnrollSchema(ma.SQLAlchemyAutoSchema):
+    
+    eng_id = ma.Int()
+    course_id = ma.Int()
+    start_date = ma.Date()
+    end_date = ma.Date()
+    has_passed = ma.Boolean()
```
Comment (yarkhinephyo) [inline restful_api/myapi/api/schemas/official_enroll.py:11]: I think these lines are unnecessary if you declare `OfficialEnrol` under `Meta`

File: restful_api/myapi/api/resources/official_enroll.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    """
+    
+  method_decorators = [jwt_required()]
+  
+  def __init__(self):
+    self.schema = OfficialEnrollSchema()
+    
+  def get(self, eng_id, course_id):
+    enrollment_record = OfficialEnroll.query.get_or_404(eng_id, course_id)
+    return {"employee": self.schema.dump(enrollment_record)}
```
Comment (yarkhinephyo) [inline restful_api/myapi/api/resources/official_enroll.py:76]: Add `msg`?

File: restful_api/myapi/manage.py (hunk 1 above)
Comment (yarkhinephyo) [inline restful_api/myapi/manage.py:21]: Did you mean to comment out?

File: restful_api/tests/factories.py (hunk 2 above)
Comment (yarkhinephyo) [inline restful_api/tests/factories.py:35]: Did you mean to comment out?

Comment (brandontjd) [reply restful_api/tests/factories.py:35]: Comment out for now, might test it in the future!

File: restful_api/myapi/manage.py (hunk 1 above)
Comment (brandontjd) [reply restful_api/myapi/manage.py:21]: Yup, see comment above ^

File: restful_api/tests/test_official_enroll.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+
+from flask import url_for
+
+def test_get_all_enrollments(client, db, official_enroll_factory, test_user_headers):
+    official_enrollment_url = url_for('api.official_enroll', eng_id=1)
+    official_enrollments = official_enroll_factory.create_batch(5)
+
```
Comment (yarkhinephyo) [inline restful_api/tests/test_official_enroll.py:7]: Super minor but a less hardcoded way might be:

```
official_enrollments = official_enroll_factory.create_batch(5)
official_enrollment_url = url_for('api.official_enroll', eng_id=official_enrollments[0].eng_id)
```

File: restful_api/myapi/api/schemas/official_enroll.py (hunk 3 above)
Comment (brandontjd) [reply restful_api/myapi/api/schemas/official_enroll.py:11]: it made a difference during testing. Without these lines, the endpoint returned `start_date` , `end_date` and `has_passed` only. Hm we KIV, I try fix in the future? Lacking full understanding of marshmallow now

File: restful_api/myapi/api/resources/official_enroll.py (hunk 4 above)
Comment (brandontjd) [reply restful_api/myapi/api/resources/official_enroll.py:76]: Fixed, thanks for catching.

File: restful_api/tests/test_official_enroll.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+from flask import url_for
+
+def test_get_all_enrollments(client, db, official_enroll_factory, test_user_headers):
```
Comment (yarkhinephyo) [inline restful_api/tests/test_official_enroll.py:4]: If you want can add a `404` no engineer scenerio and engineer found but no course scenerio. But don't have to.

File: restful_api/myapi/api/schemas/official_enroll.py (hunk 3 above)
Comment (yarkhinephyo) [reply restful_api/myapi/api/schemas/official_enroll.py:11]: Oh okay maybe my understanding off. Leave it as it is bah

**Review by yarkhinephyo — APPROVED** [review]

**Pull request merged**